tests/integration/automated_chatflows/test_solutions.py

```diff
@@ -0,0 +1,301 @@
+import string
+import sys
+from unittest import TestCase
+from uuid import uuid4
+import os
+from gevent import sleep
+from time import time
+import base_test as base
+from redis import Redis
+
+sdk_location = __file__.split("tests")[0]
+solution_automation_location = f"{sdk_location}solutions_automation"
+sys.path.append(solution_automation_location)
+import pytest
+from jumpscale.loader import j
+from solutions_automation import deployer
+
+
+@pytest.mark.integration
+class AutomatedChatflows(TestCase):
+    @classmethod
+    def setUpClass(cls):
+
+        """ Create Network"""
+        cls.network_name = base.random_string()
+        cls.wg_name = base.random_string()
+        network = deployer.create_network(solution_name=cls.network_name)
+        import pdb
+
+        pdb.set_trace()
+        _, wireguard, _ = j.sals.process.execute("sudo wg")
+        if cls.wg_name in wireguard:
+            j.sals.process.execute(f"sudo wg-quick down /tmp/{cls.wg_name}.conf")
+
+        j.sals.fs.write_file(f"/tmp/{cls.wg_name}.conf", network.wgconf)
+        # j.sals.process.execute(f'echo "{network.wgconf}" > /tmp/"{cls.wg_name}".conf')
+        j.sals.process.execute(f"sudo wg-quick up /tmp/{cls.wg_name}.conf")
+
+        # prepare ssh
+        cls.ssh_client = base.random_string()
+        if not j.sals.fs.exists("/tmp/.ssh"):
+            j.core.executors.run_local('mkdir /tmp/.ssh && ssh-keygen -t rsa -f /tmp/.ssh/id_rsa -q -N "" ')
+        cls.ssh_cl = j.clients.sshkey.get(cls.ssh_client)
+        cls.ssh_cl.private_key_path = "/tmp/.ssh/id_rsa"
+        cls.ssh_cl.load_from_file_system()
+        cls.ssh_cl.save()
+
+    @classmethod
+    def tearDownClass(cls):
+        # should stop threebot server.
+        j.sals.process.execute(f"sudo wg-quick down /tmp/{cls.wg_name}.conf")
+        j.clients.sshkey.delete(cls.ssh_client)
+        j.clients.sshclient.delete(cls.ssh_client)
+        j.sals.fs.rmtree(path=f"/tmp/{cls.wg_name}.conf")
+
+    def tearDown(self):
+        j.sals.reservation_chatflow.solutions.cancel_solution_by_uuid(self.solution_uuid)
```

Comment:
In case of the test doesn't have a `solution_uuid` like pool one, this will raise attribute error.
You may initialize this on `setUp` and check for it before canceling the solution.